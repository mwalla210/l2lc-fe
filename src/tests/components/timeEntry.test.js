import React from 'react'
import renderer from 'react-test-renderer'
import TimeEntry from '../../components/timeEntry'
jest.mock('../../components/form')

const defaultOptions = {
  website: {
    currentUser: {
      stationID: 1
    }
  },
  page: {
    formModel: {
      submissionConfirmOpen: false,
      closeModal: jest.fn(),
      errorModalOpen: false,
      errorResponse: 'errorResponse',
      setValue: jest.fn(),
      submit: jest.fn(),
    },
    setNullContent: jest.fn(),
    projectTimeEntryMenuItem: jest.fn()
  }
}

jest.useFakeTimers()

describe('TimeEntry', () => {
  it ('Renders', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <TimeEntry {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Calls onChange', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <TimeEntry {...options}/>,
    )
    const inst = component.getInstance()
    inst.onChange({target: {value: '%'}})
    expect(inst.props.page.formModel.setValue).toHaveBeenCalled()
  })
  it ('Calls onClick', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <TimeEntry {...options}/>,
    )
    const inst = component.getInstance()
    inst.onClick({preventDefault: jest.fn()})
    expect(inst.props.page.formModel.submit).toHaveBeenCalled()
  })
  it ('Calls onCancel', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <TimeEntry {...options}/>,
    )
    const inst = component.getInstance()
    inst.onCancel({preventDefault: jest.fn()})
    expect(inst.props.page.setNullContent).toHaveBeenCalled()
    expect(setTimeout).toHaveBeenCalledTimes(1)
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 100)
    jest.runAllTimers()
    expect(inst.props.page.projectTimeEntryMenuItem).toHaveBeenCalled()
  })
})
