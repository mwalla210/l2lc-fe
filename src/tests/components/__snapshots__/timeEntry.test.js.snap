// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeEntry Renders 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-6"
  >
    <p>
      Scan all projects and employees first, then scan station.
    </p>
    <form>
      <textarea
        autoFocus={true}
        className="form-control"
        onChange={[Function]}
        rows="5"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
        value={undefined}
      />
      <p>
        Project ID: undefined
      </p>
      <p>
        Employee ID: undefined
      </p>
      <p>
        Station: undefined
      </p>
      <div
        className="row justify-content-center"
      >
        <div
          className="btn-toolbar"
          role="toolbar"
        >
          <div
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-outline-secondary"
              disabled={undefined}
              onClick={[Function]}
              style={undefined}
              type="button"
            >
              Clear
            </button>
            <button
              className="btn btn-primary"
              disabled={undefined}
              onClick={[Function]}
              style={undefined}
              type="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="col-10"
  >
    <textarea
      className="form-control"
      disabled={true}
      rows="11"
      style={
        Object {
          "marginTop": "10px",
        }
      }
      value={undefined}
    />
  </div>
</div>
`;
